### PR TITLE
Implement `[setup.dictionaries]` support

### PR DESCRIPTION
**Pull request by Integralist**
The following screenshots are based on this modified rust starter-kit configuration:

```toml
# This file describes a Fastly Compute@Edge package. To learn more visit:
# https://developer.fastly.com/reference/fastly-toml/

authors = ["<oss@fastly.com>"]
description = "A basic starter kit that demonstrates routing, simple synthetic responses and overriding caching rules."
language = "rust"
manifest_version = 2
name = "Default starter for Rust"
service_id = ""

[setup.backends.backend_name]
address = "reqbin.com"
description = "A backend able to serve `/articles` path"
port = 443
[setup.backends.other_backend_name]
address = "httpbin.org"
description = "A backend able to serve `/anything` path"
port = 443

[setup.dictionaries.dict_a]
description = "My first dictionary"
[setup.dictionaries.dict_a.items.foo]
value = "my default value for foo"
description = "a good description about foo"
[setup.dictionaries.dict_a.items.bar]
value = "my default value for bar"
description = "a good description about bar"
[setup.dictionaries.dict_a.items.baz]
value = "my default value for baz"
description = "a good description about baz"

[setup.dictionaries.dict_b]
description = "My second dictionary"
[setup.dictionaries.dict_b.items.beep]
value = "my default value for beep"
description = "a good description about beep"
[setup.dictionaries.dict_b.items.boop]
value = "my default value for boop"
description = "a good description about boop"
[setup.dictionaries.dict_b.items.blah]
value = "my default value for blah"
description = "a good description about blah"
```

Screenshot 2021-10-07 at 17 19 01

Screenshot 2021-10-07 at 17 19 19

### CODE REVIEW

**Comment by Integralist** [comment]
@kailan @doramatadora @triblondon 

I'm testing the idea of adding some colour to the prompts to help distinguish them. If you look at the original screenshots in the PR description you'll see that all that white text (even though we already have bolding of prompts) just gets to be too much noise, so I've tried the following...

Screenshot 2021-10-07 at 18 05 53

**Comment by Integralist** [comment]
Here is the configuration I'm testing against...

Screenshot 2021-10-11 at 17 38 40

Here is the output of running the publish command against this configuration...

Screenshot 2021-10-11 at 17 39 03
Screenshot 2021-10-11 at 17 39 49

File: pkg/commands/compute/deploy.go [hunk 1]
```diff
@@ -171,11 +184,24 @@ func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
 	}
 
 	if newService {
+		// NOTE: A service can't be activated without at least one backend defined.
+		// This explains why the following block of code isn't wrapped in a call to
+		// the .Predefined() method, as the call to .Configure() will ensure the
+		// user is prompted regardless of whether there is a [setup.backends]
+		// defined in the fastly.toml configuration.
```
Comment (doramatadora) [inline pkg/commands/compute/deploy.go:191]: Comments like these are fantastic 🙌🏼

File: pkg/commands/compute/deploy.go [hunk 2]
```diff
@@ -210,10 +236,11 @@ func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
...
 
 	if newService {
-		// NOTE: We can't pass a text.Progress instance to setup.Backends at the
-		// point of constructing the backends object, as the text.Progress instance
-		// prevents other stdout from being read.
+		// NOTE: We can't pass a text.Progress instance to setup.Backends or
+		// setup.Dictionaries at the point of constructing the setup objects,
+		// as the text.Progress instance prevents other stdout from being read.
 		backends.Progress = progress
+		dictionaries.Progress = progress
```
Comment (doramatadora) [inline pkg/commands/compute/deploy.go:243]: Is this the same progress instance as `backends.Progress`?

File: pkg/commands/compute/deploy.go (hunk 1 above)
Comment (doramatadora) [inline pkg/commands/compute/deploy.go:191]: Comments like these are fantastic 🙌🏼

**Review by doramatadora — APPROVED** [review]
This looks good to me. I added a non-blocking question for my own understanding.

**Comment by Integralist** [comment]
**NOTE**: CI failed recently on Windows test run. GitHub failed to say why it failed. Running full test suite locally on macOS is passing and the macOS + Linux CI jobs also passed. I've triggered a re-run of all the jobs so we'll see if the Windows job fails again this time.

**Pull request merged**